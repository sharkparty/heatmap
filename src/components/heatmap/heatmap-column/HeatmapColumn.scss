.heatmap-column {
  flex: 1;
  text-align: center;
  font-size: .75em;
  text-transform: uppercase;

  .row-key {
    padding: 5px 0;
  }
}
