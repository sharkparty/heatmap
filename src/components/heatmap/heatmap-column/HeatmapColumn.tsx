// React
import * as React from 'react';
// Vendor
import * as moment from 'moment';
// Components
import HeatmapRow from '../heatmap-row/HeatmapRow';
// Interfaces
import { hoursInDay } from '../Heatmap';
// Assets
import './HeatmapColumn.css';

export interface HeatmapColumnParams {
    dayOfWeek: number | string;
    config: {};
    meta: { range: number; };
    dayData?: {};
}

export const HeatmapColumn: React.SFC<HeatmapColumnParams> = ({
    dayOfWeek,
    dayData = [],
    config,
    meta
}) => {
    return (
        <div title={moment().day(dayOfWeek).format('dddd')} className="heatmap-column">
            {hoursInDay.map((hour) => (
                <HeatmapRow
                    key={hour}
                    config={config}
                    intervalOfTime={hour}
                    intervalData={dayData[hour]}
                    meta={meta}
                />
            ))}
            <div className="row-key">
                {moment().day(dayOfWeek).format('ddd')}
            </div>
        </div>
    );
};
