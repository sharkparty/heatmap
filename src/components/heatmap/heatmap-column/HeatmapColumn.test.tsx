// React
import * as React from 'react';
// Vendor
import { shallow } from 'enzyme';
// Search
import { HeatmapColumn } from './HeatmapColumn';

describe('Component: HeatmapColumn', () => {
    it('renders without crashing', () => {
        const wrapper = shallow(<HeatmapColumn config={{}} meta={{}} dayData={{}} dayOfWeek={1} />);
        expect(wrapper).toMatchSnapshot();
    });
});
