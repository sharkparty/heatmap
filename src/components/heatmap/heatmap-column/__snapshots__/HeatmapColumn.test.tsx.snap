// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: HeatmapColumn renders without crashing 1`] = `
<div
  className="heatmap-column"
  title="Monday"
>
  <withProps(Component)
    config={Object {}}
    intervalOfTime={0}
    key="0"
    meta={Object {}}
  />
  <withProps(Component)
    config={Object {}}
    intervalOfTime={1}
    key="1"
    meta={Object {}}
  />
  <withProps(Component)
    config={Object {}}
    intervalOfTime={2}
    key="2"
    meta={Object {}}
  />
  <withProps(Component)
    config={Object {}}
    intervalOfTime={3}
    key="3"
    meta={Object {}}
  />
  <withProps(Component)
    config={Object {}}
    intervalOfTime={4}
    key="4"
    meta={Object {}}
  />
  <withProps(Component)
    config={Object {}}
    intervalOfTime={5}
    key="5"
    meta={Object {}}
  />
  <withProps(Component)
    config={Object {}}
    intervalOfTime={6}
    key="6"
    meta={Object {}}
  />
  <withProps(Component)
    config={Object {}}
    intervalOfTime={7}
    key="7"
    meta={Object {}}
  />
  <withProps(Component)
    config={Object {}}
    intervalOfTime={8}
    key="8"
    meta={Object {}}
  />
  <withProps(Component)
    config={Object {}}
    intervalOfTime={9}
    key="9"
    meta={Object {}}
  />
  <withProps(Component)
    config={Object {}}
    intervalOfTime={10}
    key="10"
    meta={Object {}}
  />
  <withProps(Component)
    config={Object {}}
    intervalOfTime={11}
    key="11"
    meta={Object {}}
  />
  <withProps(Component)
    config={Object {}}
    intervalOfTime={12}
    key="12"
    meta={Object {}}
  />
  <withProps(Component)
    config={Object {}}
    intervalOfTime={13}
    key="13"
    meta={Object {}}
  />
  <withProps(Component)
    config={Object {}}
    intervalOfTime={14}
    key="14"
    meta={Object {}}
  />
  <withProps(Component)
    config={Object {}}
    intervalOfTime={15}
    key="15"
    meta={Object {}}
  />
  <withProps(Component)
    config={Object {}}
    intervalOfTime={16}
    key="16"
    meta={Object {}}
  />
  <withProps(Component)
    config={Object {}}
    intervalOfTime={17}
    key="17"
    meta={Object {}}
  />
  <withProps(Component)
    config={Object {}}
    intervalOfTime={18}
    key="18"
    meta={Object {}}
  />
  <withProps(Component)
    config={Object {}}
    intervalOfTime={19}
    key="19"
    meta={Object {}}
  />
  <withProps(Component)
    config={Object {}}
    intervalOfTime={20}
    key="20"
    meta={Object {}}
  />
  <withProps(Component)
    config={Object {}}
    intervalOfTime={21}
    key="21"
    meta={Object {}}
  />
  <withProps(Component)
    config={Object {}}
    intervalOfTime={22}
    key="22"
    meta={Object {}}
  />
  <withProps(Component)
    config={Object {}}
    intervalOfTime={23}
    key="23"
    meta={Object {}}
  />
  <div
    className="row-key"
  >
    Mon
  </div>
</div>
`;
